src/wms/src/main/java/org/geoserver/wms/map/OpenLayers2MapOutputFormat.java

```diff
@@ -0,0 +1,54 @@
+/* (c) 2017 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.wms.map;
+
+import org.geoserver.wms.WMS;
+import org.geoserver.wms.WMSMapContent;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Subclass of {@link OpenLayersMapOutputFormat} allowing to explictly request a OpenLayers 2 client
+ */
+public class OpenLayers2MapOutputFormat extends OpenLayersMapOutputFormat {
+
+    /**
+     * Format name for OL2 preview
+     */
+    public static final String OL2_FORMAT = "application/openlayers2";
+
+    /**
+     * The mime type for the response header
+     */
+    public static final String MIME_TYPE = "text/html; subtype=openlayers2";
+
+
+    /**
+     * The formats accepted in a GetMap request for this producer and stated in getcaps
+     */
+    private static final Set<String> OUTPUT_FORMATS = new HashSet<>(Arrays.asList(OL2_FORMAT, MIME_TYPE));
+
+
+    public OpenLayers2MapOutputFormat(WMS wms) {
+        super(wms);
+    }
+
+    @Override
+    public Set<String> getOutputFormatNames() {
+        return OUTPUT_FORMATS;
+    }
+
+    @Override
+    public String getMimeType() {
+        return MIME_TYPE;
+    }
+
+    @Override
+    protected boolean isOL3Enabled(WMSMapContent mapContent) {
```

Comment:
This is a bit weird now.
Is the property to disable OL3 still in place, and if so, would that result in two openlayers map output formats that produce OL2 output?

A more consistent approach that still maintains backwards compatibility would be to have an ol2 format, and ol3 format and a default ol format that delegates to one of ol2 or ol3 based on the value of the parameter.

Reply:
I can go for that, but there is a couple of gotchas
* The OL3 format is not supported on all browsers, currently the code automatically falls back on OL2. But does it make sense when someone is explicitly asking for 3?
* OL2 is the odd one out and needs a special name because it offers some unique advantages but say we start advertising OL3 and then upgrade to OL4... does that mean we also keep 3 and add a new explicit OL4 format? From your comment above it seems there would be no reason to keep OL3 around, but removing it would break backwards compatibility.

Reply:
* If someone explicitly asks for 3, but is on a browser that doesn't support it, some sort of failure message seems appropriate (probably just a standard ServiceException).
* With respect to OL3 vs Ol4, good point. I want to say we should count OL3 and OL4 as the same thing, but that doesn't seem especially viable here. Perhaps support both? The minified js files are pretty small relative to GeoServer.

Reply:
Ok, so you're proposing to have one more subclass specific to OL3, and maybe make the base class a little bit more generic (so that adding OL4 would just be a matter of overriding a few template methods)?

Reply:
Yes, something like that seems the best option if we are going to have version-specific subclasses.